Modules/ITS/src/ITSClusterTask.cxx

```diff
@@ -153,6 +138,9 @@ void ITSClusterTask::startOfCycle()
 void ITSClusterTask::monitorData(o2::framework::ProcessingContext& ctx)
 {
 
+  mGeom = o2::its::GeometryTGeo::Instance();
+  mGeom->fillMatrixCache(o2::math_utils::bit2Mask(o2::math_utils::TransformType::L2G));
+
```

Comment:
is there a reason to move these two lines in `monitorData` instead of `initialize` (where they were before)? The same for the ITSFhrTask.cxx

Reply:
good point, that does not seem optimal

Reply:
The geometry is passed only when the task calls "run", which for the QC is the monitorData. Unfortunately, I don't see a better way. You can protect this: if already done, don't redo it. Assuming the geometry does not change. 
I would let you take care once the PR is merged.

Reply:
Note that with the DPL CCDB fetcher the geometry will not be loaded until the 1st TF processing. And the methods above to need TGeometry loaded. 
As far as I understand, this means that they should be executed in the monitor. 
Another question is that they should be executed only once in the beginning of the run, i.e. should be changed to 
```
if (pc.services().get<o2::framework::TimingInfo>().globalRunNumberChanged) {
   mGeom = o2::its::GeometryTGeo::Instance();
   mGeom->fillMatrixCache(o2::math_utils::bit2Mask(o2::math_utils::TransformType::L2G));
}
```

Reply:
Hello @shahor02 , 
Exactly, this is what I meant. I did not know about the way to check that the run number is changed, I would have checked with a `bool` :-)
Implemented now, please check.
For the B field, it is different though, since for that the DP processing is continuously ongoing. 
Chiara

Reply:
Thanks a lot @shahor02, @chiarazampolli.

Reply:
@chiarazampolli can you please apply the same also for the ITSClusterTask.cxx?

Reply:
I did actually, but I forgot to add it :-) Now done.